### PR TITLE
Lake DSL files

### PR DESCRIPTION
**Pull request by kavyasreedhar**
This PR includes the files needed for the Lake DSL and associated compiler and modular SMT collateral generation for the DSL hardware.

aha, garnet branches: collateral_clean

### CODE REVIEW

File: lake/dsl/dsl_examples/simba.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# input activation buffer
+input_buffer = TopLake(64, 1, 1)
+
+port = MemPort(1, 0)
+
+input_params = make_params("input_buffer", 1024, read_write_port_width=1, use_macro=True)
+input_buffer.add_memory(input_params, read_write_ports=[port])
+
+input_buffer.construct_lake("simba_input.sv")
```
Comment (mbstrange2) [inline lake/dsl/dsl_examples/simba.py:31]: We should talk about this sometime

File: lake/modules/cfg_reg_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                newvar = self.var_from_def(port, pname)
+                self.wire(newvar, port)
+
+                @always_ff((posedge, "clk"), (negedge, "rst_n"))
+                def add_to_config_reg(self):
+                    if ~self._rst_n:
+                        newvar = 0
+                    elif self._config_en:
+                        newvar = 1
+
```
Comment (mbstrange2) [inline lake/modules/cfg_reg_wrapper.py:77]: What does this do?

File: lake/top/lake_top.py [hunk 3]
```diff
@@ -1076,7 +1076,7 @@ def get_static_bitstream(self,
         # handling arrays in the interface.
         # To alleviate this, we create the flattened wrapper so we can query widths of config
         # registers and trim values to their bitwidths...
-        print(f"Current_name: {self.name}")
+        # print(f"Current_name: {self.name}")
```
Comment (mbstrange2) [inline lake/top/lake_top.py:1079]: Can just nuke this, was only for debugging I believe.

File: lake/utils/gen-btor.ys [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+# This turns all undriven signals into
+# inputs
+setundef -undriven -expose;
+
+#initialize all signals after reset to 0
+#sim -clock clk -resetn rst_n -n 3 -w -zinit;
+#sim -clock clk -resetn rst_n -n 2 -rstlen 2 -w -zinit;
+
+# This writes to a file in BTOR2 format
+write_btor ./global_buffer.btor2
```
Comment (mbstrange2) [inline lake/utils/gen-btor.ys:54]: Can we just get rid of all the commented commands?

File: lake/utils/util.py [hunk 5]
```diff
@@ -187,6 +219,52 @@ def extract_formal_annotation(generator, filepath, module_attr="agg"):
         print(sram_tb_shared, file=open("sram_tb_shared.txt", "w+"))
 
 
+def extract_formal_annotation_colat(generator, filepath, mem_names, edges):
```
Comment (mbstrange2) [inline lake/utils/util.py:222]: Huge nitpick, but shouldn't it be `collat`?

File: lake/utils/util.py [hunk 6]
```diff
@@ -345,7 +423,8 @@ def safe_wire(gen, w_to, w_from):
     '''
     # Only works in one dimension...
     if w_to.width != w_from.width:
-        print(f"SAFEWIRE: WIDTH MISMATCH: {w_to.name} width {w_to.width} <-> {w_from.name} width {w_from.width}")
+        if lake_util_verbose_trim:
+            print(f"SAFEWIRE: WIDTH MISMATCH: {w_to.name} width {w_to.width} <-> {w_from.name} width {w_from.width}")
```
Comment (mbstrange2) [inline lake/utils/util.py:427]: Thank you lol.

**Review by mbstrange2 — COMMENTED** [review]
Just a few comments, but once addressed this looks good to merge.

File: lake/dsl/dsl_examples/simba.py (hunk 1 above)
Comment (kavyasreedhar) [reply lake/dsl/dsl_examples/simba.py:31]: Agreed, the function/arg decisions were fairly arbitrary, so we can/should definitely change them to whatever we think is nicest.

File: lake/modules/cfg_reg_wrapper.py (hunk 2 above)
Comment (kavyasreedhar) [reply lake/modules/cfg_reg_wrapper.py:77]: I think you wrote this file lol I'm not sure

File: lake/top/lake_top.py (hunk 3 above)
Comment (kavyasreedhar) [reply lake/top/lake_top.py:1079]: Deleted.

File: lake/utils/util.py (hunk 5 above)
Comment (kavyasreedhar) [reply lake/utils/util.py:222]: Lol yes, fixed thanks

File: lake/utils/gen-btor.ys (hunk 4 above)
Comment (kavyasreedhar) [reply lake/utils/gen-btor.ys:54]: I removed this file, Nestan has this in another repo and I had it here for testing, so better not to duplicate it.

**Review by mbstrange2 — APPROVED** [review]
LGTM
